internal/pkg/service/buffer/storage/local/writer/notify/notify.go

```diff
@@ -0,0 +1,38 @@
+package notify
+
+import (
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+	"sync"
+)
+
+// Notifier allows multiple listeners to Wait for the completion of an operation that may end with an error.
+// It is used while awaiting synchronization, see more in syncWriter.
+type Notifier struct {
```

Comment:
V "najprisnejsom" rezime, potvrdime operaciu zapisu ako uspesnu, az ked su data fyciky na disku.

To znamena, ze 1..N zapisov musi cakat na synchronizaciu na disk.

Teda jeden event potrebujeme broadcastovat k N receivers.

Je viacero moznosti ako to spravit, napr `sync.Cond`, ale najjednoduchsie je pouzit `channel` a ten zavriet ked operacia skonci. Citanie zo zavreteho ch skonci vzdy hned s prazdnou hodnotou, je to bezny pattern:

`Broadcast (1-to-N) notifications by closing a channel`:
https://go101.org/article/channel-use-cases.html

- https://stackoverflow.com/questions/52035390/why-using-chan-struct-when-wait-something-done-not-chan-interface
- https://gist.github.com/betandr/a629f50e58696b85032c6c95408e3fe8